platform/qt/src/qmapboxgl.cpp

```diff
@@ -176,80 +199,146 @@ QMapboxGLSettings::QMapboxGLSettings()
+*/
 QMapboxGLSettings::ConstrainMode QMapboxGLSettings::constrainMode() const
 {
     return m_constrainMode;
 }
 
+/*!
+    Sets the map constrain \a mode.
+*/
 void QMapboxGLSettings::setConstrainMode(ConstrainMode mode)
 {
     m_constrainMode = mode;
 }
 
+/*!
+    Returns the viewport mode. This is used to flip the vertical
+    orientation of the map as some devices may use inverted orientation.
+
+    By default, it is set to QMapboxGLSettings::DefaultViewport.
+*/
 QMapboxGLSettings::ViewportMode QMapboxGLSettings::viewportMode() const
 {
     return m_viewportMode;
 }
 
+/*!
+    Sets the viewport \a mode.
+*/
 void QMapboxGLSettings::setViewportMode(ViewportMode mode)
 {
     m_viewportMode = mode;
 }
 
+/*!
+    Returns the cache database maximum hard size in bytes. The database
+    will grow until the limit is reached. Setting a maximum size smaller
+    than the current size of an existing database is undefined.
+
+    By default, it is set to 50 MB.
+*/
 unsigned QMapboxGLSettings::cacheDatabaseMaximumSize() const
 {
     return m_cacheMaximumSize;
 }
 
+/*!
+    Returns the maximum allowed cache database \a size in bytes.
+*/
 void QMapboxGLSettings::setCacheDatabaseMaximumSize(unsigned size)
 {
     m_cacheMaximumSize = size;
 }
 
+/*!
+    Returns the cache database path. The cache is used for storing
+    recently used resources like tiles and also an offline tile database
+    pre-populated by the
+    \l {https://github.com/mapbox/mapbox-gl-native/blob/master/bin/offline.sh}
+    {Offline Tool}.
+
+    By default, it is set to \b :memory: meaning it will create a in-memory
+    cache instead of a file in the disk.
+*/
 QString QMapboxGLSettings::cacheDatabasePath() const
 {
     return m_cacheDatabasePath;
 }
 
+/*!
+    Sets the cache database \a path.
+
+    Setting the \a path to \b :memory: will create an in-memory cache.
+*/
 void QMapboxGLSettings::setCacheDatabasePath(const QString &path)
 {
     m_cacheDatabasePath = path;
 }
 
+/*!
+    Returns the asset path, which is the root directory from where
+    the \b asset:// scheme gets resolved in a style. \b asset:// can be used
+    for loading a resource from the disk in a style rather than fetching
+    it from the network.
+
+    By default, it is set to the value returned by QCoreApplication::applicationDirPath().
+*/
 QString QMapboxGLSettings::assetPath() const
 {
     return m_assetPath;
 }
 
+/*!
+    Sets the asset \a path.
+*/
 void QMapboxGLSettings::setAssetPath(const QString &path)
 {
     m_assetPath = path;
 }
 
+/*!
+    Returns the access token.
+
+    By default, it is taken from the environment variable \b MAPBOX_ACCESS_TOKEN
+    or empty if the variable is not set.
+*/
 QString QMapboxGLSettings::accessToken() const {
     return m_accessToken;
 }
 
+/*!
+    Sets the access \a token. An access token is required for styles and tiles
+    hosted at Mapbox servers.
+
+    An access token for the Starter plan can be created free of charge at the
```

Comment:
The iOS and macOS SDKs link to https://www.mapbox.com/help/define-access-token/ and https://www.mapbox.com/studio/account/tokens/ and also [frame](https://www.mapbox.com/ios-sdk/api/3.4.0-beta.6/Classes/MGLAccountManager.html#/c:objc(cs)MGLAccountManager(cm)setAccessToken:) the mechanism in terms that the developer can relate to.